allocator.go

```diff
@@ -61,198 +28,80 @@ type options struct {
 	withMinimalChurn bool
 	// maxChurn limits the number of moves needed to fulfill an allocation request with respect to a prior allocation.
 	maxChurn int64
-	// prevAssignment holds some prior allocation.
-	prevAssignment map[rangeId][]nodeId
 	// searchTimeout forces the solver to return within the specified duration.
 	searchTimeout time.Duration
 	// verboseLogging routes all the internal solver logs to stdout.
 	verboseLogging bool
 }
 
-// Option manifests a closure that mutates allocation configurations in accordance with caller preferences.
-type Option func(*options)
+// AllocOption manifests a closure that mutates allocation configurations in accordance with caller preferences.
+type AllocOption func(*allocOptions)
 
-// Allocator holds the ranges, nodes, underlying CP-SAT solver, assigment variables, and configuration needed.
-type Allocator struct {
-	// ranges are a mapping of rangeId onto the ckRange struct.
-	ranges map[rangeId]*ckRange
-	// nodes are a mapping of nodeId onto the node struct.
-	nodes map[nodeId]*node
+// allocator holds the replicas, nodes, underlying CP-SAT solver, assigment variables, and configuration needed.
+type allocator struct {
+	*ClusterState
```

Comment:
Can you add a quick comment on this field? (Similar to how the ones beneath is have small comments)

Reply:
Any reason for using this as an anonymous (no named) field?